specification/trace/semantic_conventions/messaging.md

```diff
@@ -234,21 +239,34 @@ The following operations related to messages are defined for these semantic conv
 
 **[3]:** If the span describes an operation on a batch of messages.
 
-**[4]:** Only if span represents operation on a single message.
+**[4]:** If value is `true`. When missing, the value is assumed to be `false`.
 
-**[5]:** Only for spans that represent an operation on a single message.
+**[5]:** Destination name SHOULD uniquely identify a specific queue, topic or other entity within the broker. If
+the broker does not have such notion, the destination name SHOULD uniquely identify the broker.
+
+**[6]:** If span describes operation on a single message or if the value applies to all messages in the batch.
+
+**[7]:** Destination names could be constructed from templates. An example would be a destination name involving a user name or product id. Although the destination name in this case is of high cardinality, the underlying template is of low cardinality and can be effectively used for grouping and aggregation.
+
+**[8]:** If available. Instrumentations MUST NOT use `messaging.destination.name` as template unless low-cardinality of destination name is guaranteed.
+
+**[9]:** If value is `true`. When missing, the value is assumed to be `false`.
 
-**[6]:** Only if span represents operation on a single message.
+**[10]:** Only if span represents operation on a single message.
 
-**[7]:** Only if span represents operation on a single message.
+**[11]:** Only for spans that represent an operation on a single message.
 
-**[8]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+**[12]:** Only if span represents operation on a single message.
 
-**[9]:** This should be the IP/hostname of the broker (or other network-level peer) this specific message is sent to/received from.
+**[13]:** Only if span represents operation on a single message.
 
-**[10]:** Typically observed from the client side, and represents a proxy or other intermediary domain name.
+**[14]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
 
-**[11]:** If different than `server.address` and if `server.socket.address` is set.
+**[15]:** This should be the IP/hostname of the broker (or other network-level peer) this specific message is sent to/received from.
+
+**[16]:** Typically observed from the client side, and represents a proxy or other intermediary domain name.
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/issues/1925 😢

Reply:
Yeah.. maybe it's time to really consider this.